`flatbuffers` fails to compile on macOS v 11.4

Version information
--------------------
ddlog - `f68ee83c44`
Java - `v16.0.1`
Flatbuffers - `v1.11.0`
Cmake - `v3.20.5`
Compiler - `AppleClang 12.0.5.12050022`

Log:
```
+ make -j
[  3%] Building CXX object CMakeFiles/flathash.dir/src/flathash.cpp.o
[  4%] Building CXX object CMakeFiles/flatbuffers.dir/src/code_generators.cpp.o
[  4%] Building CXX object CMakeFiles/flatbuffers.dir/src/idl_parser.cpp.o
[  6%] Building CXX object CMakeFiles/flatc.dir/src/code_generators.cpp.o
[  7%] Building CXX object CMakeFiles/flatbuffers.dir/src/idl_gen_text.cpp.o
[  9%] Building CXX object CMakeFiles/flatc.dir/src/idl_parser.cpp.o
[ 11%] Building CXX object CMakeFiles/flatbuffers.dir/src/reflection.cpp.o
[ 12%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_text.cpp.o
[ 15%] Building CXX object CMakeFiles/flatc.dir/src/reflection.cpp.o
[ 15%] Building CXX object CMakeFiles/flatbuffers.dir/src/util.cpp.o
[ 17%] Building CXX object CMakeFiles/flatc.dir/src/util.cpp.o
[ 19%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_cpp.cpp.o
[ 22%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_dart.cpp.o
[ 22%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_general.cpp.o
[ 23%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_js_ts.cpp.o
[ 25%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_go.cpp.o
[ 26%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_php.cpp.o
[ 28%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_python.cpp.o
[ 30%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_lobster.cpp.o
[ 31%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_lua.cpp.o
[ 33%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_rust.cpp.o
[ 34%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_grpc.cpp.o
[ 36%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_fbs.cpp.o
[ 38%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_json_schema.cpp.o
[ 39%] Building CXX object CMakeFiles/flatc.dir/src/flatc_main.cpp.o
[ 41%] Building CXX object CMakeFiles/flatc.dir/src/flatc.cpp.o
[ 44%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/cpp_generator.cc.o
[ 44%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/go_generator.cc.o
[ 46%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/java_generator.cc.o
[ 47%] Linking CXX executable flathash
[ 47%] Built target flathash
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::Object>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Object>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:2730:26: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Object>' requested here
  auto objs__ = builder_.CreateVectorOfSortedTables(&object_offsets);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::Object>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Object>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Object>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:2730:26: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Object>' requested here
  auto objs__ = builder_.CreateVectorOfSortedTables(&object_offsets);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::Enum>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Enum>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:2731:26: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Enum>' requested here
  auto enum__ = builder_.CreateVectorOfSortedTables(&enum_offsets);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::Enum>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Enum>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Enum>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:2731:26: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Enum>' requested here
  auto enum__ = builder_.CreateVectorOfSortedTables(&enum_offsets);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::Service>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Service>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:2734:26: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Service>' requested here
  auto serv__ = builder_.CreateVectorOfSortedTables(&service_offsets);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::Service>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Service>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Service>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:2734:26: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Service>' requested here
  auto serv__ = builder_.CreateVectorOfSortedTables(&service_offsets);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::Field>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Field>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:2781:26: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Field>' requested here
  auto flds__ = builder->CreateVectorOfSortedTables(&field_offsets);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::Field>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::Field>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Field>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:2781:26: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::Field>' requested here
  auto flds__ = builder->CreateVectorOfSortedTables(&field_offsets);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::KeyValue>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::KeyValue>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:3050:21: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::KeyValue>' requested here
    return builder->CreateVectorOfSortedTables(&attrs);
                    ^
In file included from /Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:23:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:25:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1720:25: error: definition of implicit copy constructor for 'TableKeyComparator<reflection::KeyValue>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    TableKeyComparator &operator=(const TableKeyComparator &);
                        ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1735:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::TableKeyComparator<reflection::KeyValue>' first required here
    std::sort(v, v + len, TableKeyComparator<T>(buf_));
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1749:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::KeyValue>' requested here
    return CreateVectorOfSortedTables(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/src/idl_parser.cpp:3050:21: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedTables<reflection::KeyValue>' requested here
    return builder->CreateVectorOfSortedTables(&attrs);
                    ^
5 errors generated.
5 errors generated.
make[2]: *** [CMakeFiles/flatc.dir/src/idl_parser.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....none
```

`v2.0.0` compiles fine.
 `v1.12.0` results in a different error

```
+ make -j
[  1%] Building CXX object CMakeFiles/flatbuffers.dir/src/idl_parser.cpp.o
[  2%] Building CXX object CMakeFiles/flathash.dir/src/flathash.cpp.o
[  3%] Building CXX object CMakeFiles/flatbuffers.dir/src/idl_gen_text.cpp.o
[  4%] Building CXX object CMakeFiles/flatbuffers.dir/src/reflection.cpp.o
[  5%] Building CXX object CMakeFiles/flatbuffers.dir/src/util.cpp.o
[  7%] Building CXX object CMakeFiles/flatc.dir/src/idl_parser.cpp.o
[  8%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_text.cpp.o
[  9%] Building CXX object CMakeFiles/flatc.dir/src/reflection.cpp.o
[ 10%] Building CXX object CMakeFiles/flatc.dir/src/util.cpp.o
[ 12%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_csharp.cpp.o
[ 12%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_cpp.cpp.o
[ 14%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_dart.cpp.o
[ 15%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_kotlin.cpp.o
[ 16%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_go.cpp.o
[ 17%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_java.cpp.o
[ 18%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_js_ts.cpp.o
[ 20%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_php.cpp.o
[ 21%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_python.cpp.o
[ 22%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_rust.cpp.o
[ 23%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_grpc.cpp.o
[ 25%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_json_schema.cpp.o
[ 25%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_swift.cpp.o
[ 28%] Building CXX object CMakeFiles/flatc.dir/src/flatc.cpp.o
[ 29%] Building CXX object CMakeFiles/flatc.dir/src/code_generators.cpp.o
[ 30%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/python_generator.cc.o
[ 30%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/swift_generator.cc.o
[ 31%] Building CXX object CMakeFiles/flatc.dir/src/flatc_main.cpp.o
[ 32%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/java_generator.cc.o
[ 34%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/cpp_generator.cc.o
[ 35%] Building CXX object CMakeFiles/flatc.dir/grpc/src/compiler/go_generator.cc.o
[ 36%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_fbs.cpp.o
[ 38%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_lua.cpp.o
[ 38%] Building CXX object CMakeFiles/flatc.dir/src/idl_gen_lobster.cpp.o
[ 40%] Linking CXX executable flathash
[ 40%] Built target flathash
[ 41%] Linking CXX static library libflatbuffers.a
[ 41%] Built target flatbuffers
[ 42%] Linking CXX executable flatc
[ 42%] Built target flatc
[ 43%] Run generation: 'tests/monster_test_generated.h'
[ 44%] Run generation: 'tests/monster_test.bfbs'
[ 45%] Run generation: 'samples/monster.bfbs'
[ 47%] Run generation: 'tests/namespace_test/namespace_test2_generated.h'
[ 48%] Run generation: 'tests/namespace_test/namespace_test1_generated.h'
[ 49%] Run generation: 'tests/union_vector/union_vector_generated.h'
[ 50%] Run generation: 'tests/arrays_test_generated.h'
[ 51%] Run generation: 'tests/native_type_test_generated.h'
[ 54%] Run generation: 'tests/arrays_test.bfbs'
[ 54%] Run generation: 'tests/monster_test_bfbs_generated.h'
[ 55%] Run generation: 'tests/monster_extra_generated.h'
[ 56%] Run generation: 'samples/monster_generated.h'
[ 57%] All generated files were updated.
[ 57%] Built target generated_code
[ 58%] Building CXX object CMakeFiles/flatsampletext.dir/src/reflection.cpp.o
[ 60%] Building CXX object CMakeFiles/flatsampletext.dir/src/util.cpp.o
[ 61%] Building CXX object CMakeFiles/flatsampletext.dir/src/idl_parser.cpp.o
[ 62%] Building CXX object CMakeFiles/flatsampletext.dir/src/idl_gen_text.cpp.o
[ 63%] Building CXX object CMakeFiles/flattests.dir/src/idl_parser.cpp.o
[ 64%] Building CXX object CMakeFiles/flatsampletext.dir/samples/sample_text.cpp.o
[ 65%] Building CXX object CMakeFiles/flatsamplebfbs.dir/src/idl_parser.cpp.o
[ 67%] Building CXX object CMakeFiles/flattests.dir/src/reflection.cpp.o
[ 68%] Building CXX object CMakeFiles/flattests.dir/src/idl_gen_text.cpp.o
[ 69%] Building CXX object CMakeFiles/flatsamplebinary.dir/samples/sample_binary.cpp.o
[ 70%] Building CXX object CMakeFiles/flatsamplebfbs.dir/src/idl_gen_text.cpp.o
[ 71%] Building CXX object CMakeFiles/flattests.dir/src/util.cpp.o
[ 72%] Building CXX object CMakeFiles/flatsamplebfbs.dir/src/reflection.cpp.o
[ 74%] Building CXX object CMakeFiles/flattests.dir/tests/test.cpp.o
[ 75%] Building CXX object CMakeFiles/flattests.dir/src/idl_gen_fbs.cpp.o
[ 76%] Building CXX object CMakeFiles/flatsamplebfbs.dir/src/util.cpp.o
[ 77%] Building CXX object CMakeFiles/flattests.dir/tests/test_assert.cpp.o
[ 78%] Building CXX object CMakeFiles/flattests.dir/tests/test_builder.cpp.o
[ 80%] Building CXX object CMakeFiles/flattests.dir/tests/native_type_test_impl.cpp.o
[ 81%] Building CXX object CMakeFiles/flatsamplebfbs.dir/samples/sample_bfbs.cpp.o
[ 82%] Building CXX object CMakeFiles/flattests.dir/src/code_generators.cpp.o
[ 83%] Linking CXX executable flatsamplebinary
[ 84%] Built target flatsamplebinary
In file included from /Users/none/differential-datalog/flatbuffers/tests/test_builder.cpp:1:
In file included from /Users/none/differential-datalog/flatbuffers/tests/test_builder.h:7:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1799:26: error: definition of implicit copy constructor for 'StructKeyComparator<MyGame::Example::Ability>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    StructKeyComparator &operator=(const StructKeyComparator &);
                         ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1839:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::StructKeyComparator<MyGame::Example::Ability>' first required here
    std::sort(v, v + len, StructKeyComparator<T>());
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1812:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedStructs<MyGame::Example::Ability>' requested here
    return CreateVectorOfSortedStructs(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/tests/monster_test_generated.h:2083:67: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedStructs<MyGame::Example::Ability>' requested here
  auto testarrayofsortedstruct__ = testarrayofsortedstruct ? _fbb.CreateVectorOfSortedStructs<MyGame::Example::Ability>(testarrayofsortedstruct) : 0;
                                                                  ^
1 error generated.
make[2]: *** [CMakeFiles/flattests.dir/tests/test_builder.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /Users/none/differential-datalog/flatbuffers/tests/test.cpp:19:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:28:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/reflection.h:365:8: error: definition of implicit copy constructor for 'pointer_inside_vector<flatbuffers::Table, unsigned char>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  void operator=(const pointer_inside_vector &piv);
       ^
/Users/none/differential-datalog/flatbuffers/tests/test.cpp:971:16: note: in implicit copy constructor for 'flatbuffers::pointer_inside_vector<flatbuffers::Table, unsigned char>' first required here
  auto rroot = flatbuffers::piv(
               ^
In file included from /Users/none/differential-datalog/flatbuffers/tests/test.cpp:19:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:28:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/reflection.h:365:8: error: definition of implicit copy constructor for 'pointer_inside_vector<flatbuffers::Vector<unsigned char>, unsigned char>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  void operator=(const pointer_inside_vector &piv);
       ^
/Users/none/differential-datalog/flatbuffers/tests/test.cpp:980:21: note: in implicit copy constructor for 'flatbuffers::pointer_inside_vector<flatbuffers::Vector<unsigned char>, unsigned char>' first required here
  auto rinventory = flatbuffers::piv(
                    ^
In file included from /Users/none/differential-datalog/flatbuffers/tests/test.cpp:19:
In file included from /Users/none/differential-datalog/flatbuffers/include/flatbuffers/idl.h:28:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/reflection.h:365:8: error: definition of implicit copy constructor for 'pointer_inside_vector<flatbuffers::Vector<flatbuffers::Offset<flatbuffers::String>>, unsigned char>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  void operator=(const pointer_inside_vector &piv);
       ^
/Users/none/differential-datalog/flatbuffers/tests/test.cpp:994:29: note: in implicit copy constructor for 'flatbuffers::pointer_inside_vector<flatbuffers::Vector<flatbuffers::Offset<flatbuffers::String>>, unsigned char>' first required here
  auto rtestarrayofstring = flatbuffers::piv(
                            ^
In file included from /Users/none/differential-datalog/flatbuffers/tests/test.cpp:33:
/Users/none/differential-datalog/flatbuffers/tests/monster_test_generated.h:1914:19: error: definition of implicit copy constructor for 'MonsterBuilder' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  MonsterBuilder &operator=(const MonsterBuilder &);
                  ^
/Users/none/differential-datalog/flatbuffers/tests/test.cpp:3008:26: note: in implicit copy constructor for 'MyGame::Example::MonsterBuilder' first required here
  auto monster_builder = MonsterBuilder(builder);
                         ^
In file included from /Users/none/differential-datalog/flatbuffers/tests/test.cpp:18:
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1799:26: error: definition of implicit copy constructor for 'StructKeyComparator<MyGame::Example::Ability>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
    StructKeyComparator &operator=(const StructKeyComparator &);
                         ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1839:27: note: in implicit copy constructor for 'flatbuffers::FlatBufferBuilder::StructKeyComparator<MyGame::Example::Ability>' first required here
    std::sort(v, v + len, StructKeyComparator<T>());
                          ^
/Users/none/differential-datalog/flatbuffers/include/flatbuffers/flatbuffers.h:1812:12: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedStructs<MyGame::Example::Ability>' requested here
    return CreateVectorOfSortedStructs(data(*v), v->size());
           ^
/Users/none/differential-datalog/flatbuffers/tests/monster_test_generated.h:2083:67: note: in instantiation of function template specialization 'flatbuffers::FlatBufferBuilder::CreateVectorOfSortedStructs<MyGame::Example::Ability>' requested here
  auto testarrayofsortedstruct__ = testarrayofsortedstruct ? _fbb.CreateVectorOfSortedStructs<MyGame::Example::Ability>(testarrayofsortedstruct) : 0;
                                                                  ^
5 errors generated.
make[2]: *** [CMakeFiles/flattests.dir/tests/test.cpp.o] Error 1
[ 85%] Linking CXX executable flatsamplebfbs
make[1]: *** [CMakeFiles/flattests.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 87%] Linking CXX executable flatsampletext
[ 88%] Built target flatsamplebfbs
[ 89%] Built target flatsampletext
make: *** [all] Error 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 - Closures that depend on generic types generate invalid Rust (#1072).
 
+### FlatBuffers
+
+- Upgrade to FlatBuffers v2.0.0.  The previous version of FlatBuffers used in
+  DDlog is not compatible with recent OS X releases.
+  
+
 ## [0.48.2] - Sep 13, 2021
 
 ### Bug fixes

--- a/lib/ddlog_std.flatbuf.rs
+++ b/lib/ddlog_std.flatbuf.rs
@@ -70,6 +70,7 @@ impl<'a, T, F> FromFlatBuffer<fbrt::Vector<'a, F>> for ddlog_std::Vec<T>
 where
     T: Ord + FromFlatBuffer<F::Inner>,
     F: fbrt::Follow<'a> + 'a,
+    <F as fbrt::Follow<'a>>::Inner: Debug,
 {
     fn from_flatbuf(fb: fbrt::Vector<'a, F>) -> ::std::result::Result<Self, String> {
         let mut vec = ddlog_std::Vec::with_capacity(fb.len());
@@ -110,6 +111,7 @@ impl<'a, T, F> FromFlatBuffer<fbrt::Vector<'a, F>> for ddlog_std::Set<T>
 where
     T: Ord + FromFlatBuffer<F::Inner>,
     F: fbrt::Follow<'a> + 'a,
+    <F as fbrt::Follow<'a>>::Inner: Debug,
 {
     fn from_flatbuf(fb: fbrt::Vector<'a, F>) -> ::std::result::Result<Self, String> {
         let mut set = ddlog_std::Set::new();
@@ -152,6 +154,7 @@ where
 impl<'a, K, V, F> FromFlatBuffer<fbrt::Vector<'a, F>> for ddlog_std::Map<K, V>
 where
     F: fbrt::Follow<'a> + 'a,
+    <F as fbrt::Follow<'a>>::Inner: Debug,
     K: Ord,
     ddlog_std::tuple2<K, V>: FromFlatBuffer<F::Inner>,
 {

--- a/lib/hashset.flatbuf.rs
+++ b/lib/hashset.flatbuf.rs
@@ -25,6 +25,7 @@ impl<'a, T, F> FromFlatBuffer<fbrt::Vector<'a, F>> for crate::typedefs::hashset:
 where
     T: ::core::hash::Hash + ::core::cmp::Eq + ::core::clone::Clone + FromFlatBuffer<F::Inner>,
     F: fbrt::Follow<'a> + 'a,
+    <F as fbrt::Follow<'a>>::Inner: Debug,
 {
     fn from_flatbuf(
         fb: fbrt::Vector<'a, F>,

--- a/lib/tinyset.flatbuf.rs
+++ b/lib/tinyset.flatbuf.rs
@@ -27,6 +27,7 @@ impl<'a, T, F> FromFlatBuffer<fbrt::Vector<'a, F>> for Set64<T>
 where
     T: Ord + FromFlatBuffer<F::Inner> + u64set::Fits64,
     F: fbrt::Follow<'a> + 'a,
+    <F as fbrt::Follow<'a>>::Inner: Debug,
 {
     fn from_flatbuf(fb: fbrt::Vector<'a, F>) -> Result<Self, String> {
         let mut set = Set64::new();

--- a/rust/template/Cargo.toml
+++ b/rust/template/Cargo.toml
@@ -38,8 +38,8 @@ phf = { version = "0.10.0", features = ["macros"] }
 
 # FlatBuffers dependency enabled by the `flatbuf` feature.
 # flatbuffers crate version must be in sync with the flatc compiler and Java
-# libraries: flatbuffers "0.6" <-> FlatBuffers "1.11.0".
-flatbuffers = { version = "0.6", optional = true }
+# libraries: flatbuffers "2.0.0" <-> FlatBuffers "2.0.0".
+flatbuffers = { version = "2.0.0", optional = true }
 
 [dependencies.differential_datalog]
 path = "./differential_datalog"

--- a/rust/template/src/flatbuf.rs
+++ b/rust/template/src/flatbuf.rs
@@ -10,7 +10,7 @@ use enum_primitive_derive::Primitive;
 use flatbuffers::{self as fbrt, FlatBufferBuilder, Follow, Push, Vector, WIPOffset};
 use num_traits::FromPrimitive;
 use ordered_float::OrderedFloat;
-use std::{collections::BTreeSet, hash::Hash};
+use std::{collections::BTreeSet, fmt::Debug, hash::Hash};
 
 /// Trait for types that can be de-serialized from FlatBuffer-embedded objects.
 pub trait FromFlatBuffer<T>: Sized {
@@ -306,12 +306,18 @@ impl<'a> ToFlatBufferVectorElement<'a> for String {
 
 /// Iterator based on `fbrt::Vector`
 #[derive(Debug)]
-pub struct FBIter<'a, F> {
+pub struct FBIter<'a, F: flatbuffers::Follow<'a> + 'a>
+where
+    <F as flatbuffers::Follow<'a>>::Inner: Debug,
+{
     vec: Vector<'a, F>,
     off: usize,
 }
 
-impl<'a, F: 'a> FBIter<'a, F> {
+impl<'a, F: flatbuffers::Follow<'a> + 'a> FBIter<'a, F>
+where
+    <F as flatbuffers::Follow<'a>>::Inner: Debug,
+{
     pub fn from_vector(vec: Vector<'a, F>) -> FBIter<'a, F> {
         FBIter { vec, off: 0 }
     }
@@ -320,6 +326,7 @@ impl<'a, F: 'a> FBIter<'a, F> {
 impl<'a, F> Iterator for FBIter<'a, F>
 where
     F: flatbuffers::Follow<'a> + 'a,
+    <F as flatbuffers::Follow<'a>>::Inner: Debug,
 {
     type Item = F::Inner;
 
@@ -543,7 +550,7 @@ impl FlatbufConverter for DDlogFlatbufConverter {
     }
 
     fn query_index_from_buffer(&self, buffer: &[u8]) -> Result<(IdxId, DDValue), String> {
-        let q = flatbuffers::get_root::<fb::__Query<'_>>(buffer);
+        let q = unsafe { flatbuffers::root_unchecked::<fb::__Query<'_>>(buffer) };
 
         if let Some(key) = q.key() {
             Ok((
@@ -563,7 +570,7 @@ impl FlatbufConverter for DDlogFlatbufConverter {
         }
 
         // Each record takes at least 8 bytes of FlatBuffer space.
-        let mut fbb = fbrt::FlatBufferBuilder::new_with_capacity(8 * delta_size);
+        let mut fbb = fbrt::FlatBufferBuilder::with_capacity(8 * delta_size);
         let mut cmds = Vec::with_capacity(delta_size);
 
         for (relid, rel) in delta.iter() {
@@ -597,7 +604,7 @@ impl FlatbufConverter for DDlogFlatbufConverter {
 pub fn updates_from_flatbuf<'a>(
     buf: &'a [u8],
 ) -> Response<FBIter<'a, fbrt::ForwardsUOffset<fb::__Command<'a>>>> {
-    if let Some(cmds) = fb::get_root_as___commands(buf).commands() {
+    if let Some(cmds) = unsafe { fb::root_as___commands_unchecked(buf) }.commands() {
         Ok(FBIter::from_vector(cmds))
     } else {
         Err("Invalid buffer: failed to extract commands".to_string())
@@ -611,7 +618,7 @@ where
     let size = vals.size_hint().0;
 
     /* Each value takes at least 4 bytes of FlatBuffer space. */
-    let mut fbb = fbrt::FlatBufferBuilder::new_with_capacity(4 * size);
+    let mut fbb = fbrt::FlatBufferBuilder::with_capacity(4 * size);
     let mut val_tables = Vec::with_capacity(size);
 
     for val in vals {

--- a/test/datalog_tests/ovn_ftl.dl
+++ b/test/datalog_tests/ovn_ftl.dl
@@ -5,7 +5,7 @@
 // -Flow: the output table
 // -Logical_Switch
 
-typedef stage = LS_IN_ACL
+typedef stage_t = LS_IN_ACL
               | LS_OUT_ACL
               | LR_IN_ADMISSION
               | LR_IN_IP_INPUT
@@ -36,7 +36,7 @@ typedef stage = LS_IN_ACL
               | LS_OUT_PORT_SEC_IP
               | LS_OUT_PORT_SEC_L2
 
-output relation Flow(lr: bigint, stage: stage, prio: bigint, matchStr: string, actionStr: string)
+output relation Flow(lr: bigint, stage: stage_t, prio: bigint, matchStr: string, actionStr: string)
 input relation Logical_Router(lr: bigint)
 input relation Logical_Router_Port(name: string, lr: bigint, mac: string, enabled: bool)
 

--- a/test/datalog_tests/tutorial.dat
+++ b/test/datalog_tests/tutorial.dat
@@ -181,9 +181,9 @@ echo Evens:;
 dump Evens;
 
 start;
-insert Vector(["a", "--", "b"], "--"),
-insert Vector(["--", "a", "b"], "--"),
-insert Vector(["a", "b", "--"], "--"),
+insert Vect(["a", "--", "b"], "--"),
+insert Vect(["--", "a", "b"], "--"),
+insert Vect(["a", "b", "--"], "--"),
 commit;
 
 echo Prefix:;

--- a/test/datalog_tests/tutorial.dl
+++ b/test/datalog_tests/tutorial.dl
@@ -292,10 +292,10 @@ function prefixBefore(vec: Vec<'A>, v: 'A): Vec<'A> {
     res
 }
 
-input relation Vector(vec: Vec<string>, sep: string)
+input relation Vect(vec: Vec<string>, sep: string)
 output relation Prefix(vec: Vec<string>)
 
-Prefix(prefixBefore(vec, sep)) :- Vector(vec, sep).
+Prefix(prefixBefore(vec, sep)) :- Vect(vec, sep).
 
 /*
  * Example: Multiple heads
@@ -689,8 +689,8 @@ input relation Load_Balancer (
     name:         string
 )
 
-typedef stage = LS_IN_PRE_LB
-              | LS_OUT_PRE_LB
+typedef stage_t = LS_IN_PRE_LB
+                | LS_OUT_PRE_LB
 
 input relation Logical_Switch (
     ls:  bigint
@@ -698,7 +698,7 @@ input relation Logical_Switch (
 
 // Relation that represents OVS flows.
 output relation Flow(lr: bigint,
-                     stage: stage,
+                     stage: stage_t,
                      prio: bigint,
                      matchStr: string,
                      actionStr: string)
@@ -735,7 +735,7 @@ Flow(.lr=ls,
 
 // Introduce another Flow relation, so that we can compare the results of the two encodings.
 output relation Flow1(lr: bigint,
-                      stage: stage,
+                      stage: stage_t,
                       prio: bigint,
                       matchStr: string,
                       actionStr: string)

--- a/tools/install-flatbuf.sh
+++ b/tools/install-flatbuf.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-FLATBUF_VERSION="1.11.0"
+FLATBUF_VERSION="2.0.0"
 set -ex
 
 retry() {
@@ -9,13 +9,11 @@ retry() {
 }
 
 fetch_flatbuf_unix() {
-    rm -rf flatbuffers
-    mkdir flatbuffers
-    curl -L https://github.com/google/flatbuffers/archive/v${FLATBUF_VERSION}.tar.gz | tar -zx -C flatbuffers --strip-components 1
+    git clone https://github.com/google/flatbuffers.git --branch v${FLATBUF_VERSION}
 }
 
 fetch_flatbuf_windows() {
-    curl -L https://github.com/google/flatbuffers/releases/download/v${FLATBUF_VERSION}/flatc_windows_exe.zip > fb.zip && unzip fb.zip
+    curl -L https://github.com/google/flatbuffers/releases/download/v${FLATBUF_VERSION}/Windows.flatc.binary.zip > fb.zip && unzip fb.zip
 }
 
 echo "Installing Flatbuf"
@@ -36,7 +34,7 @@ if [ "x`flatbuffers/flatc --version`" != "xflatc version ${FLATBUF_VERSION}" ]; 
         cd ..
     else
         retry fetch_flatbuf_unix
-        # On Windows, fetch pre-build executable instead of compiling from source.
+        # On Windows, fetch pre-built executable instead of compiling from source.
         retry fetch_flatbuf_windows
     fi
 fi

